---
title: "Analysis Report 1: Amplicons and the Letter Q"
author: "Derek Newberger and help from Maddie Shehan"
date: "November 11, 2018"
output: github_document
bibliography: references.bib
csl: bioinformatics.csl
---

# Introduction

Add about 1.5-2 pages here. Must cite at least 10 peer reviewed articles.

Expanding technologies beckons the attempt to answer large
questions with big data. Originally the data acquired for
this question involved developing the forensic technique of
using unique microbiomes left by fingers on a keyborad to
link them to the owner of the keyboard. However, with such a
large dataset, there are other questions raised that require
some analyisis as to what could have caused the oserved
results. My hypothesis is that the most frequently used keys on a keyborad would have the greatest sequence abundance. However, as observed by viewing a bar graph presented in class, this does not hold consistenly ture. Therefore, the major inconsistency to be assesed is how the Qkey has essentially double the sequence abundance than the Ukey, when the Qkey is used far less and in the very least. Hopefully this will shed some light as to how a key with a loer frequency of use has a much greater sequence abundance. A series of assumptions have to be made and the variables that perhaps influence either frequency of contact will be assesesed in the following. 

Which hand uses the key
  Proportion of right handed individuals
  Do dominate hands are dirtier?
  In hand washing what often gets neglected?
Type of finger used per key
  Is a particular finger dirty?
  Is there a finger that is less used?
Type of key
  Frequency of use of keys (in the English language)
  Which row is the key on?
  Could multiple fingers touch it?
This question is important because...
  Could be applied to hand washing... knowing least contaminated finger, which finger to swab for another one of these studies. There is a difference. Sequencing is expensive, so can focus on one finger. handshake with right hand. 


# Methods

## Sample origin and sequencing

For the Fierer et al. sample collection, they swabbed each
individual key of three personal computer keyboards (n=25-30
keys per board) for ten seconds each. The swabs were
cotton-tipped and previously autoclaved that were
pre-moistened with a sterile solution. The skin of the
ventral surface of the distal joint of each fingertip of the
owner of the keyboard was also swabbed. These computers and
their respective keyboards were primarily only used by the
respective owner.
When the individuals were sampled, they were healthy and had
refrained from taking antibiotics in the past 6 months and
were 20-35 years old. The keyboards were left untouched for
half an hour before being swabbed and within ten minutes of
that the fingers were swabbed. They took samples from 15
other space bars as to compare them to the initial sampling.
All swab samples were stored at -80 C for less than 1 week
before DNA extraction.
For the DNA extraction, genomic DNA was extracted from the
swabs using the MO BIO PowerSoil DNA Isolation kit. Each
sample was frozen, incubated, shaken (MO BIO vortex adapter),
amplified (using previous optimized primer set), placed in a
PCR with reverse primers of HotMasterMix (5 PRIME),
denatured, replicated with amplicons which were then pooled
and visualized on agarose gels using SYBR Safe DNA gel stain.
The amplicons wee then cleaned with the UltraClean-htp
96-well PCR Clean-up kit. Amplicon DNA concentrations were
measured using the Quant-iTPicoGreendsDNAreagentandkit. They
precipitated the final pool of DNA on ice and solutions and
centrifuged it. The supernatant pellet was washed,
centrifuged, then air dried and then resuspended in nuclease
free water. Pyrosequencing was carried out on a 454 Life
Sciences Genome Sequencer FLX instrument by the Environmental
Genomics Core Facility at the University of South Carolina.
They believe to have found that this process can potentially
be used as a forensic metric since the communities matched up
well with the corresponding individuals swabbed.


## Computational

These are the methods you used. Should probably be at least a half of a page. At a very minimum should include citations for DADA2 [@callahan2016] and phyloseq [@mcmurdie2013]. Note that these don't count towards the five references you need to cite in the introduction. We will use the Ribosomal Database Project reference taxonomy to assign taxonomy to unkown sequences [@cole2008ribosomal].

### Starting the project and pre-existing code

In order to retireve this dataset I forked and cloned the repository to my github DerekNewberger-analysis-report-01-amplicons from the class repository. Using RStudio I modified the Analysis_Report_01_amplicons.Rmd file where there were pre-existing chunks of code. The code installed the necessary libraries (dplyr, tidyr, knitr, ggplot2, citr, dada2, seqinr, mctoolsr, and phyloseq). The sequences longer than 300 base pairs were filtered and trimmed since the quality apparead to suffere after 200 base pairs. We used build error models from each of the samples to learn of the possible error rates. Using derepfastq we removed any sequences that were duplicates which removed around half of the sequences we had remaining in the dataset. Using dada and recommendations found on github or 454 data we adjusted the parameters and gave the HOMOPOLYMER_GAP_PENALTY a -1 since 454 sequencing has a difficulty dealing with this aspect of sequencing. A ‘site by species matrix’ was made, and it formed a table that instead of species, used the actual sequence as the “species” name. We then searched for and then removed chimeras using removeBimeraDenovo. Since the sampling method used was swabbing different surfaces and not an isolation DNA extraction technique, there is a chance that multiple sequences from different organisms were fused together. We built a phylogeny using the cleaned data and assigned them a taxonomy to each sequence variant based on a supplied training set of known sequences. There is code that can build a phylogenetic tree using a phyloseq object from dada2 outputs. 

### Constructing the code for the figures


# Results

include sufficient text in this section to describe what your findings were. Remember that in the results section you just describe what you found
```{r load-libraries, message = FALSE}
# Be sure to install these packages before running this script
# They can be installed either with the intall.packages() function
# or with the 'Packages' pane in RStudio

# load general-use packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")

# this package allows for the easy inclusion of literature citations in our Rmd
# more info here: https://github.com/crsh/citr
# and here:
# http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html
library("citr")

# These are the primary packages well use to clean and analyze the data
# this package needs to be installed from bioconductor -- it's not on CRAN
# see info here: https://benjjneb.github.io/dada2/dada-installation.html
library("dada2")

# This to export a fasta of our final denoised sequence variants
library("seqinr")

# To install this you have to install from GitHub
# See more info here: https://github.com/leffj/mctoolsr
# run this -- install.packages("devtools")
# and then this -- devtools::install_github("leffj/mctoolsr")
library("mctoolsr")

# And this to visualize our results
# it also needs to be installed from bioconductor
library("phyloseq")
```

```{r extract-sample-and-file-names}
# NOTE: Much of the following follows the DADA2 tutorials available here:
# https://benjjneb.github.io/dada2/tutorial.html
# Accessed October 19, 2017

# set the base path for our input data files
path <- "data/raw_data"

# Sort ensures samples are in order
filenames_forward_reads <- sort(list.files(path, pattern = ".fastq"))

# Extract sample names, assuming filenames have format: SAMPLENAME.fastq
sample_names <- sapply(strsplit(filenames_forward_reads, "\\."), `[`, 1)

# Specify the full path to each of the filenames_forward_reads
filenames_forward_reads <- file.path(path, filenames_forward_reads)
```


We can see from the quality profiles that most reads tend to get pretty bad in quality after around 200 bases. 

```{r filter-reads, include = FALSE}
# Place filtered files in filtered/ subdirectory
# note this will fail if the directory doesn't exist
filter_path <- file.path("output", "filtered")
filtered_reads_path <- file.path(filter_path,
                                 paste0(sample_names,
                                        "_filt.fastq.gz"))

# See ?filterAndTrim for details on the parameters
# See here for adjustments for 454 data:
# https://benjjneb.github.io/dada2/
#     faq.html#can-i-use-dada2-with-my-454-or-ion-torrent-data
filtered_output <- filterAndTrim(fwd = filenames_forward_reads,
                                 filt = filtered_reads_path,
                                 maxLen = 300,
                                 maxN = 0, # discard any seqs with Ns
                                 maxEE = 3, # allow w/ up to 3 expected errors
                                 truncQ = 2, # cut off if quality gets this low
                                 rm.phix = TRUE,
                                 compress = TRUE,
                                 multithread = FALSE)
```

```{r learn-errors}
# this build error models from each of the samples
errors_forward_reads <- learnErrors(filtered_reads_path,
                                   multithread = FALSE)
```

```{r dereplicate-sequences}
# get rid of any duplicated sequences
dereplicated_forward_reads <- derepFastq(filtered_reads_path,
                                         verbose = TRUE)

# Name the derep-class objects by the sample names
names(dereplicated_forward_reads) <- sample_names
```

```{r run-dada}
# parameters adjusted based on recommendations for 454 data here:
# https://benjjneb.github.io/dada2/
#     faq.html#can-i-use-dada2-with-my-454-or-ion-torrent-data
dada_forward_reads <- dada(dereplicated_forward_reads,
                           err = errors_forward_reads,
                           HOMOPOLYMER_GAP_PENALTY = -1, # reduce penalty bc 454
                           BAND_SIZE = 32) # performs local alignments bc indels

# check dada results
# dada_forward_reads
```
```{r make-sequence-table}
# produce the 'site by species matrix'
sequence_table <- makeSequenceTable(dada_forward_reads)
```

```{r remove-chimeras}
# Check for and remove chimeras
sequence_table_nochim <- removeBimeraDenovo(sequence_table,
                                            method = "consensus",
                                            multithread = FALSE,
                                            verbose = TRUE)

# What percent of our reads are non-chimeric?
non_chimeric_reads <- round(sum(sequence_table_nochim) / sum(sequence_table),
                            digits = 4) * 100
```

After removing chimeras, we were left with `r non_chimeric_reads`% of our cleaned reads.

```{r extract-sequences-to-fasta}
# we want to export the cleaned, trimmed, filtered, denoised sequence variants
# so that we can build a phylogeny - we'll build the phylogeny outside of R
# but we need the fasta file to do so. We keep the names of each sequence as the
# sequence itself (which is rather confusing), because that's how DADA2 labels
# it's columns (e.g. 'species')
# function taken from https://github.com/benjjneb/dada2/issues/88
export_taxa_table_and_seqs <- function(sequence_table_nochim,
                                       file_seqtab,
                                       file_seqs) {
  seqtab_t <- as.data.frame(t(sequence_table_nochim)) # transpose to data frame
  seqs <- row.names(seqtab_t) # extract rownames
  row.names(seqtab_t) <- seqs # set rownames to sequences
  outlist <- list(data_loaded = seqtab_t)
  mctoolsr::export_taxa_table(outlist, file_seqtab) # write out an OTU table
  seqs <- as.list(seqs)
  seqinr::write.fasta(seqs, row.names(seqtab_t), file_seqs) # write out fasta
 }

# run the function, with the names of the files we want it to create
# and where to put them
export_taxa_table_and_seqs(sequence_table_nochim,
                           "output/sequence_variants_table.txt",
                           "output/sequence_variants_seqs.fa")
```

```{r assign-taxonomy}
# assigns taxonomy to each sequence variant based on a supplied training set
# made up of known sequences
 taxa <- assignTaxonomy(sequence_table_nochim,
                       "data/training/rdp_train_set_16.fa.gz",
                       multithread = FALSE,
                       tryRC = TRUE) # also check with seq reverse compliments
```

```{r read-in-metadata-and-create-phyloseq}
# read in the phylogeny, which was created from the fasta exported above
# in Geneious by aligning the sequences with MAFFT and then building a
# Maximum-Likelihood tree with RAxML
 tree_in <- read_tree("output/sequence_variants_MAFFT_FastTree.newick")

# Next we want to read in the metadata file so we can add that in too
# This is not a csv file, so we have to use a slightly different syntax
# here the `sep = "\t"` tells the function that the data are tab-delimited
# and the `stringsAsFactors = FALSE` tells it not to assume that things are
# categorical variables
metadata_in <- read.table(paste0("data/metadata/",
                    "fierer_hand_bacteria_SRA_study_ERP022626_SraRunTable.txt"),
                          sep = "\t",
                          header = TRUE,
                          stringsAsFactors = FALSE,
                          row.names = 6) # sets sample IDs to row names

# Construct phyloseq object (straightforward from dada2 outputs)
phyloseq_obj <- phyloseq(otu_table(sequence_table_nochim,
                                   taxa_are_rows = FALSE), # sample-spp matrix
                         sample_data(metadata_in), # metadata for each sample
                         tax_table(taxa), # taxonomy for each sequence variant
                         phy_tree(tree_in)) # phylogeny from sequence variants
```


```{r sample-source-abundance}
# bar plot of sample source (keyboard key or finger) and Abundance of sequences
plot_bar(phyloseq_obj, x = "sample_source",
         title = "Sequence abundance for all sample sources")
```
**Figure 1**: Abundance of sequences within the sapmle_sourse(s).
We see how the abundance of sequences on the each key are not all even.
This sparked interest on the difference of abundance between the U and
Q keys. From here a focus on the hand or finger used for each key will
be compared. 

```{r comparing-abundance-for-q-and-u-keys}
# bar plot of samples from Q and U keys and Abundance of sequences
phyloseq_obj_qu <- subset_samples(phyloseq_obj,
                                  sample_source == "Ukey" |
                                    sample_source == "Qkey")
plot_bar(phyloseq_obj_qu, x = "sample_source",
         title = "Sequence abundance on Q and U keys")
```
**Figure 2**: Abundance of sequences within the sapmle_sourse of the Ukey and
Qkey. Since in the English language there are very few words and no common
words that use the letter "Q" without "U" it was unexpected to see how the
ideally more isolated Qkey having double the amount of sequence abundance.
From this it will be invesitgated to see if left handed keys carry a higher
abundance than right keys or the type of finger used is the cause. 

```{r comparing-abundance-for-s-and-l-keys}
# bar plot of samples from L and S keys and Abundance of sequences
phyloseq_obj_sl <- subset_samples(phyloseq_obj,
                                  sample_source == "Skey" |
                                    sample_source == "Lkey")
plot_bar(phyloseq_obj_sl, x = "sample_source",
         title = "Sequence abundance on L and S keys")
```
**Figure 3**: Abundance of sequences within the sapmle_sourse of the Skey and
Lkey. To separate handedness and to keep the fingers consistent, letters S
and L were selected. It shows that the Lkey, which is on the right side of
the keyboard, has a higher abundance. The data for the keys touched by the
ringerfinger shows how either hand can led to a higher abundance. It is
important to note that the Qkey had a higher abundance eventhough it is
likely to be exposed or touched by the same finger that touches the Skey.

```{r comparing-abundance-for-g-and-h-keys}
# bar plot of samples from G and H keys and Abundance of sequences
phyloseq_obj_gh <- subset_samples(phyloseq_obj,
                                  sample_source == "Gkey" |
                                    sample_source == "Hkey")
plot_bar(phyloseq_obj_gh, x = "sample_source",
         title = "Sequence abundance on G and H keys")
```
**Figure 4**: Abundance of sequences within the sapmle_sourse of the Gkey and
Hkey. As shown, the Gkey as on the left side has a higher abundance. The
index fingers also show a higher overall abundance than the previous graphs.

```{r comparing-abundance-for-e-and-o-keys}
# bar plot of samples from E and O keys and Abundance of sequences
phyloseq_obj_eo <- subset_samples(phyloseq_obj,
                                  sample_source == "Ekey" |
                                    sample_source == "Okey")
plot_bar(phyloseq_obj_eo, x = "sample_source",
                                  title = "Sequence abundance on E and O keys")
```
**Figure 5**: Abundance of sequences within the sapmle_sourse of the Ekey and
Okey. The Ekey and the Okey were selected for they are the most frequently used
key on their respective side. Both the use and the abundance of the Ekey are
greater than the Okey. Only one of our graphs show that a key on the right side
had a greater abundance.

```{r comparing-abundance-between-left-and-right-hand-keys}
# box plot of left and right handed keys sequence abundance
phyloseq_table <- psmelt(phyloseq_obj)

phyloseq_abundance_by_key <- phyloseq_table %>%
group_by(sample_source) %>%
summarize(sumAbundance = sum(Abundance))

phyloseq_abundance_by_key$hand <-
  ifelse(phyloseq_abundance_by_key$sample_source %in%
       c("Qkey", "Wkey", "Ekey", "Rkey", "Akey", "Skey", "Dkey",
         "Fkey", "Gkey", "Zkey", "Xkey", "Ckey", "Vkey", "Tkey",
         "Bkey"), "L",
       ifelse(phyloseq_abundance_by_key$sample_source %in%
              c("Ykey", "Ukey", "Ikey", "Okey", "Pkey", "Hkey", "Jkey",
                "Lkey", "Nkey", "Mkey"), "R",
              ifelse(phyloseq_abundance_by_key$sample_source
                     == "Space_bar", "B", "NA")
              )
       )

phyloseq_abundance_by_key_lr <- phyloseq_abundance_by_key %>%
  filter(hand == "L" | hand == "R")
ggplot(phyloseq_abundance_by_key_lr, aes(y = sumAbundance, x = hand)) +
geom_boxplot() + ggtitle("Comparing abundance of sequences per hand")

```
**Figure 6**: Comparing abundance of sequences of the left and right hands.
Instead of comparing letter by letter, this box plot shows the general trend
that the left keys had a higher sequence abundance.

```{r comparing-mean-abundance-between-handed-keys}
# bar plot of mean abundance for left and right handed keys

phyloseq_abundance_by_key_lrb <- phyloseq_abundance_by_key %>%
  filter(hand == "L" | hand == "R" | hand == "B") %>%
  group_by(hand) %>%
summarize(meanSumAbundance = mean(sumAbundance))
ggplot(phyloseq_abundance_by_key_lrb, aes(x = hand,
                                              y = meanSumAbundance)) +
  geom_col() +
 ggtitle("Comparing mean abundance of sequences left and right handed keys")
```
**Figure 7**: Comparing abundance of sequences for left and right handed keys
and the space bar. The space bar has the highest mean abundance while the trend
of the left hand having a higher sequence abundance can still be observed. The
difference between the mean abundance of either hand is dwarfed when taking into
condideration the keys that are touched by both.

As shown in Figure 1, there was a difference in the abundance of sequences on the
Qkey and the Ukey. Observing an overall difference between the keys was as expected,
however, in the English language there are not many instances where one uses the
letter "q" without the letter "u", though it must be assumed that primarily only
English was used by those who perfomed this experiment in Boulder, Colorado, U.S.A.
It is important to note that this connection is apparently even more severe in
Spanish with "Qatar" being one (if not only) words in that language that use q
without u (cite). 

Figure 7- "Both" includes three other females. 
Figure 8- To separate them?
Have to assume that they type normally.
Get rid of "o" or
Make a graph that combines the others. 
# Discussion

There are multpile instances where the opposite can be obervved, but the instance that was the most notable was how the Qkey

Add around 2-3 pages interpreting your results and considering future directions one might take in analyzing these data.

# Sources Cited


